osu.Game.Rulesets.Mania/Objects/Drawables/DrawableHoldNote.cs

```diff
@@ -260,6 +272,15 @@ protected override void Update()
                 sizingContainer.Height = 1;
         }
 
+        protected override void ApplySkin(ISkinSource skin, bool allowFallback)
+        {
+            base.ApplySkin(skin, allowFallback);
+            var skinTailOrigin = skin.GetConfig<ManiaSkinConfigurationLookup, Anchor>(
+                new ManiaSkinConfigurationLookup(LegacyManiaSkinConfigurationLookups.HoldNoteTailOrigin)
+            );
+            tailOrigin = skinTailOrigin ?? new Bindable<Anchor>(Anchor.BottomCentre);
```

Comment:
Don't store bindable references from external sources, also use bindable feedback.

```diff
diff --git a/osu.Game.Rulesets.Mania/Objects/Drawables/DrawableHoldNote.cs b/osu.Game.Rulesets.Mania/Objects/Drawables/DrawableHoldNote.cs
index 04b09e9d0f..14e31a5568 100644
--- a/osu.Game.Rulesets.Mania/Objects/Drawables/DrawableHoldNote.cs
+++ b/osu.Game.Rulesets.Mania/Objects/Drawables/DrawableHoldNote.cs
@@ -32,9 +32,10 @@ public partial class DrawableHoldNote : DrawableManiaHitObject<HoldNote>, IKeyBi
 
         public IBindable<bool> IsHitting => isHitting;
 
-        private IBindable<Anchor> tailOrigin;
         private readonly Bindable<bool> isHitting = new Bindable<bool>();
 
+        private readonly Bindable<Anchor> tailOrigin = new Bindable<Anchor>(Anchor.BottomCentre);
+
         public DrawableHoldNoteHead Head => headContainer.Child;
         public DrawableHoldNoteTail Tail => tailContainer.Child;
         public DrawableHoldNoteBody Body => bodyContainer.Child;
@@ -129,6 +130,10 @@ protected override void LoadComplete()
             base.LoadComplete();
 
             isHitting.BindValueChanged(updateSlidingSample, true);
+
+            tailOrigin.BindValueChanged(_ => updateTailOrigin());
+            Direction.BindValueChanged(_ => updateTailOrigin());
+            updateTailOrigin();
         }
 
         protected override void OnApply()
@@ -242,17 +247,12 @@ protected override void Update()
             // The rationale for this is account for heads/tails with corner radius.
             bodyPiece.Y = (Direction.Value == ScrollingDirection.Up ? 1 : -1) * Head.Height / 2;
             bodyPiece.Height = DrawHeight - Head.Height / 2;
+
             if (tailOrigin.Value == Anchor.TopCentre)
                 bodyPiece.Height -= Tail.Height / 2;
             else
                 bodyPiece.Height += Tail.Height / 2;
 
-            // Update the origin of the tail piece, taking into account the scrolling direction
-            if (Direction.Value == ScrollingDirection.Up)
-                Tail.Origin = tailOrigin.Value == Anchor.TopCentre ? Anchor.BottomCentre : Anchor.TopCentre;
-            else
-                Tail.Origin = tailOrigin.Value;
-
             if (Time.Current >= HitObject.StartTime)
             {
                 // As the note is being held, adjust the size of the sizing container. This has two effects:
@@ -275,10 +275,7 @@ protected override void Update()
         protected override void ApplySkin(ISkinSource skin, bool allowFallback)
         {
             base.ApplySkin(skin, allowFallback);
-            var skinTailOrigin = skin.GetConfig<ManiaSkinConfigurationLookup, Anchor>(
-                new ManiaSkinConfigurationLookup(LegacyManiaSkinConfigurationLookups.HoldNoteTailOrigin)
-            );
-            tailOrigin = skinTailOrigin ?? new Bindable<Anchor>(Anchor.BottomCentre);
+            tailOrigin.Value = skin.GetConfig<ManiaSkinConfigurationLookup, Anchor>(new ManiaSkinConfigurationLookup(LegacyManiaSkinConfigurationLookups.HoldNoteTailOrigin))?.Value ?? Anchor.BottomCentre;
         }
 
         protected override void CheckForResult(bool userTriggered, double timeOffset)
@@ -392,6 +389,14 @@ private void updateSlidingSample(ValueChangedEvent<bool> tracking)
                 slidingSample?.Stop();
         }
 
+        private void updateTailOrigin()
+        {
+            if (Direction.Value == ScrollingDirection.Up)
+                Tail.Origin = tailOrigin.Value == Anchor.TopCentre ? Anchor.BottomCentre : Anchor.TopCentre;
+            else
+                Tail.Origin = tailOrigin.Value;
+        }
+
         protected override void OnFree()
         {
             slidingSample.ClearSamples();
```

Reply:
I found out it was due to https://github.com/ppy/osu/blob/4490dbf8960214dcc9d3f67dbfc88ab5255c6e94/osu.Game.Rulesets.Mania/Objects/Drawables/DrawableManiaHitObject.cs#L68-L71

To workaround this, I'd have to override `OnDirectionChanged` inside `DrawableHoldNoteTail` and duplicate the lookup logic, which I'd like to avoid. Is there another way to do this? Maybe make `tailOrigin` public and reference that inside the tail class?